packages/mco/hooks/protected-applications.ts

```diff
@@ -0,0 +1,190 @@
+              (drPolicyToDiscoveredAppsCount[drPolicyName] || 0) + 1;
+          }
+        });
+      }
+      return [protectedPlacementMap, drPolicyToDiscoveredAppsCount];
+    }, [drpcs, drpcLoadedWAError]);
+
+  const drPolicyToAppsetCount = React.useMemo(() => {
+    const drPolicyToAppsetCount: { [key: string]: number } = {};
+    if (argoAppSetsLoadedWAError) {
+      argoAppSets?.forEach((argoAppSet) => {
+        const placementName = findPlacementNameFromAppSet(argoAppSet);
+        const placementUniqueName = getPlacementUniqueName(
+          placementName,
+          ACMPlacementModel.kind,
+          getNamespace(argoAppSet)
+        );
+
+        if (protectedPlacementMap.hasOwnProperty(placementUniqueName)) {
+          protectedPlacementMap[placementUniqueName].forEach((drpc) => {
+            const drPolicyName = drpc.spec.drPolicyRef.name;
+            drPolicyToAppsetCount[drPolicyName] =
+              (drPolicyToAppsetCount[drPolicyName] || 0) + 1;
+          });
+        }
+      });
+    }
+    return drPolicyToAppsetCount;
+  }, [argoAppSets, protectedPlacementMap, argoAppSetsLoadedWAError]);
+
+  const drPolicyToSubsAppCount = React.useMemo(() => {
+    const drPolicyToSubsAppCount: { [key: string]: number } = {};
+    if (subsLoadedWAError) {
+      const subsAppMap: { [key: string]: Set<string> } = {};
+      subscriptions?.forEach((subscription) => {
+        if (isPlacementModel(subscription)) {
+          const applicationName = subscription?.metadata?.labels?.app;
+          const subsAppUniqueName = `${applicationName}-${getNamespace(
+            subscription
+          )}`;
+          const placementRef = subscription.spec.placement.placementRef;
+          const placementUniqueName = getPlacementUniqueName(
+            placementRef.name,
+            placementRef.kind,
+            getNamespace(subscription)
+          );
+
+          if (!subsAppMap[subsAppUniqueName]) {
+            subsAppMap[subsAppUniqueName] = new Set();
+          }
+          subsAppMap[subsAppUniqueName].add(placementUniqueName);
+        }
+      });
+
+      Object.values(subsAppMap).forEach((placementUniqueNames) => {
+        const drPolicyNames = new Set<string>();
+        placementUniqueNames.forEach((placementUniqueName) => {
+          if (protectedPlacementMap.hasOwnProperty(placementUniqueName)) {
+            const drpc = Array.from(
+              protectedPlacementMap[placementUniqueName]
+            )[0]; // Assuming we need the first DRPC if multiple
+            drPolicyNames.add(drpc.spec.drPolicyRef.name);
+          }
+        });
+        drPolicyNames.forEach((name) => {
+          drPolicyToSubsAppCount[name] =
+            (drPolicyToSubsAppCount[name] || 0) + 1;
+        });
+      });
+    }
+    return drPolicyToSubsAppCount;
+  }, [subscriptions, subsLoadedWAError, protectedPlacementMap]);
+
+  const combinedApplications = React.useMemo(() => {
+    const combined = {};
```

Comment:
```suggestion
    const drPolicyToAppCount = {};
```
And use the type for this